qiskit_experiments/framework/experiment_data.py

```diff
@@ -767,6 +770,49 @@ def add_data(
     ) -> None:
         """Add experiment data.
 
+        Args:
+            data: Experiment data to add. Several types are accepted for convenience:
+
+                * Result: Add data from this ``Result`` object.
+                * List[Result]: Add data from the ``Result`` objects.
+                * Dict: Add this data.
+                * List[Dict]: Add this list of data.
+
+        Raises:
+            TypeError: If the input data type is invalid.
+        """
+
+        if any(not future.done() for future in self._analysis_futures.values()):
+            LOG.warning(
+                "Not all analysis has finished running. Adding new data may "
+                "create unexpected analysis results."
+            )
+        if not isinstance(data, list):
+            data = [data]
+
+        # Directly add non-job data
+        with self._result_data.lock:
+            for datum in data:
+                if isinstance(datum, dict):
+                    if datum["metadata"].get("composite_metadata"):
```

Comment:
```suggestion
                    if "composite_metadata" in datum["metadata"]:
```
if value is not necessary